internal/cache/file/downloader/downloader.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package downloader
+
+import (
+	"os"
+	"path"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/cache/data"
+	"github.com/googlecloudplatform/gcsfuse/internal/cache/lru"
+	"github.com/googlecloudplatform/gcsfuse/internal/locker"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/gcs"
+)
+
+// getObjectPath gives object path which is concatenation of bucket and object
+// name separated by "/".
+func getObjectPath(bucketName string, objectName string) string {
+	return path.Join(bucketName, objectName)
+}
+
+// JobManager is responsible for maintaining, getting and removing file download
+// jobs.
+type JobManager struct {
+	fileInfoCache        *lru.Cache
+	perm                 os.FileMode
+	cacheLocation        string
+	sequentialReadSizeMb int32
+
+	jobs map[string]*Job
+	mu   locker.Locker
+}
+
+func NewJobManager(fileInfoCache *lru.Cache, perm os.FileMode, cacheLocation string, sequentialReadSizeMb int32) (jm *JobManager) {
```

Comment:
perm - is FilePerms in the ServerConfig, right?
If this is the case, then let's rename it filePerm. Also, please write a one line comment, what is it used for.

Reply:
it will not be exactly filePerms in ServerConfig. The filePerms in ServerConfig defines the permission for file in GCS bucket. This defines the permission of file in cache. We can't keep the filePerms in ServerConfig same because the file in cache will always be created by gcsfuse daemon. So we will keep a default value which restricts any other user from accessing it if allow-other is not passed at the time of mounting. If allow-other is passed then we will allow others to access it.
but the type of both are same, so I changed the name to filePerm and added documentation.